cmd/kms-handlers.go

```diff
@@ -236,7 +249,27 @@ func (a kmsAPIHandlers) KMSListKeysHandler(w http.ResponseWriter, r *http.Reques
 	}
 	manager, ok := GlobalKMS.(kms.KeyManager)
 	if !ok {
-		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrNotImplemented), r.URL)
+		// If no key found in kms, check env variable MINIO_KMS_SECRET_KEY
+		status, err := GlobalKMS.Stat(ctx)
+		if err != nil {
+			writeCustomErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrInternalError), err.Error(), r.URL)
+			return
+		}
+		if len(status.DefaultKey) != 0 {
+			secretKey := []kes.KeyInfo{
+				{
+					Name: status.DefaultKey,
+				},
+			}
+			res, err := json.Marshal(secretKey)
+			if err != nil {
+				writeCustomErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrInternalError), err.Error(), r.URL)
+				return
+			}
+			writeSuccessResponseJSON(w, res)
+			return
+		}
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
```

Comment:
Instead of this implement List() call to succeed for ENV Key, avoids this kind of superflous code.